### PR TITLE
fix: add ca opiton if https

### PR DESCRIPTION
**Pull request by ahungrynoob**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message follows commit guidelines

### CODE REVIEW

**Comment by ahungrynoob** [comment]
ci 有毒啊

**Comment by ahungrynoob** [comment]
https://github.com/eggjs/egg-cluster/pull/90

**Comment by atian25** [comment]
@ahungrynoob rebase 下解决冲突

File: lib/app_worker.js [hunk 1]
```diff
@@ -52,6 +52,9 @@ function startServer(err) {
       key: fs.readFileSync(options.https.key),
       cert: fs.readFileSync(options.https.cert),
     });
+    if (options.https.ca) {
+      httpsOptions.ca = fs.readFileSync(options.https.ca);
```
Comment (atian25) [inline lib/app_worker.js]: 直接放到上面吧，`ca: options.https.ca && fs.readFileSync(options.https.ca);`

File: lib/utils/options.js [hunk 2]
```diff
@@ -39,14 +39,20 @@ module.exports = function(options) {
...
         cert: options.cert,
       };
+      if (options.ca) {
+        options.https.ca = options.ca;
+      }
     }
     assert(options.https.key && fs.existsSync(options.https.key), 'options.https.key should exists');
     assert(options.https.cert && fs.existsSync(options.https.cert), 'options.https.cert should exists');
+    if (options.https.ca) {
+      assert(options.https.ca && fs.existsSync(options.https.ca), 'options.https.ca should exists');
```
Comment (atian25) [inline lib/utils/options.js]:
```suggestion
      assert(!options.https.ca || fs.existsSync(options.https.ca), 'options.https.ca should exists');
```

不需要 if 了

File: lib/utils/options.js [hunk 3]
```diff
@@ -39,14 +39,20 @@ module.exports = function(options) {
   if (options.https) {
     if (is.boolean(options.https)) {
       // TODO: compatible options.key, options.cert, will remove at next major
-      deprecate('[master] Please use `https: { key, cert }` instead of `https: true`');
+      deprecate('[master] Please use `https: { key, cert, ca }` instead of `https: true`');
       options.https = {
         key: options.key,
         cert: options.cert,
       };
+      if (options.ca) {
```
Comment (atian25) [inline lib/utils/options.js]: 我觉得这个可以不用支持了，毕竟我们推荐的是另一种模式，这里没必要再加个参数了，你觉得呢？

File: lib/utils/options.js (hunk 2 above)
Comment (ahungrynoob) [reply lib/utils/options.js]: 巧妙啊

**Comment by atian25** [comment]
thanks for your work. landed at 1.25.0

**Pull request merged**